docs/supervisor/development.md

```diff
@@ -23,6 +23,70 @@ When the initializing is complete you can access the Home Assistant frontend on
 
+#### Using Windows WSL2
+
+While the system is Linux and compatible with Mac, it is also possible to build and test locally on Windows using WSL2 with Debian and other similar distros. This is achieved via Windows Subsystem for Linux 2 (WSL2), which makes POSIX tasks possible within Windows via system-level integration with a Linux virtual machine.
+
+:::tip
+You must use WSL2. WSL version 1 is not capable of running Docker and therefore is incompatible. If using this method, you must ensure you have [set up WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10).
+:::
+
+It is assumed at this point you have installed Debian from the Microsoft Store and [Visual Studio Code](https://code.visualstudio.com/download). From within your WSL2-enabled Debian terminal, run the following annotated bash script. If you copy the entire script it should perform the required setup of Debian for Home Assistant development but any problems may be difficult to troubleshoot, so it is recommended to run this line-by-line.
+
+```bash
+apt update; apt install -y curl wget git gnupg; #Update and install our dependencies
+curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash; #Install NVM
+source ~/.profile; nvm install node; #reload profile and install node
+sudo apt remove cmdtest yarn  #Remove problematic packages if installed. Debian current Yarn does not work.
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -  #Add yarn repo certificate
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list #Add Yarn repo
+sudo apt update; sudo apt install yarn -y; #Install yarn
+curl -fsSL https://get.docker.com |sudo bash  #install docker Community Edition
+sudo update-alternatives --set iptables /usr/sbin/iptables-legacy #Fix IP Tables for Docker
+git clone https://github.com/home-assistant/supervisor /home/$USER/supervisor #Clone Supervisor to ~/supervisor
+usermod -aG docker $USER #Add self to docker group
+systemctl enable docker.service #enable docker start on boot
+service docker start #Start the docker service
+service docker status #Verify docker status is running
+```
+
+Restart your Debian terminal to activate the new profiles and software configurations. 
+
+:::tip
+Always run scripts line by line to figure out where the problem lies. If you need to completely start over, you can uninstall the Debian app, then from the Windows command prompt execute `del %HOMEPATH%\AppData\Roaming\Code`. After removing your VSCode settings, launch VS Code, and remove all extensions. Finally either execute `net stop LxssManager` from an administrative `cmd` prompt or reboot your computer and try again.
+:::
+
+Now you're ready to run Yarn and Node. We must set up Docker communications and launch Visual Studio Code. The following commands will need to be run after each reboot/restart to link Docker to your WSL instance and launch Visual Studio Code.
```

Comment:
I'd probably go for Docker Desktop for Windows here instead since it just works; and note that you should verify you're not required to pay Docker (which I expect applies to 99.9% of people following these instructions).

Reply:
I'm not sure Docker Desktop is preferred here.   Docker desktop is a tiered subscription, personal, team, pro, or business.   There's licensing involved here.  Regardless, Docker Desktop MAY be used, but a docker MUST be installed within WSL in order to use the WSL container and communicate with the docker running in another WSL container.

Reply:
I would say Docker Desktop is preferred unless the user knows better.

If the user is running Windows to begin with, then it's likely the user will find it easier to install and setup Docker Desktop. Otherwise the user is most likely running commands that he doesn't understand in a linux prompt and it will make troubleshooting docker more troublesome.

Reply:
@frenck explicitly forbids Docker.io due to licensing in other comments here. Docker Desktop would fall into the same category.

Reply:
Rancher Desktop from SUSE is an open source (Apache 2.0) alternative.